src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -433,15 +432,27 @@ public void CleanCache()
             {
                 var componentModel = await _componentModel.GetValueAsync(jobCts.Token);
 
-                var logger = componentModel.GetService<RestoreOperationLogger>();
-                await logger.StartAsync(
-                    request.RestoreSource,
-                    ErrorListProvider,
-                    _joinableFactory,
-                    jobCts);
-
-                var job = componentModel.GetService<ISolutionRestoreJob>();
-                return await job.ExecuteAsync(request, _restoreJobContext, logger, jobCts.Token);
+                using (var logger = componentModel.GetService<RestoreOperationLogger>())
+                {
+                    try
+                    {
+                        // Start logging
+                        await logger.StartAsync(
+                            request.RestoreSource,
+                            _errorListTableDataSource,
+                            _joinableFactory,
+                            jobCts);
+
+                        // Run restore
+                        var job = componentModel.GetService<ISolutionRestoreJob>();
+                        return await job.ExecuteAsync(request, _restoreJobContext, logger, jobCts.Token);
+                    }
+                    finally
+                    {
+                        // Complete all logging
+                        await logger.StopAsync();
```

Comment:
it will aways run from threadpool. without configureAwait, await will keep previous context it ran.